examples/interactions/param.py

```diff
@@ -63,9 +63,9 @@ async def description(
 # by using `Param` and passing a callable.
 @bot.slash_command()
 async def defaults(
-    inter: disnake.CommandInteraction,
+    inter: disnake.CommandInteraction[commands.Bot],
     string: str = "this is a default value",
-    user: disnake.User = commands.Param(lambda inter: inter.author),
+    user: disnake.User = commands.Param(lambda inter: inter.author),  # type: ignore
```

Comment:
would really love to get rid of the type ignore here, but I believe this would require us to make `commands.Param` a generic class or mess around with covariance; just leaving this here for now for discussion, I'll try to look into it if I have time

Reply:
not sure what I thought back when I wrote that, but this isn't actually all that complex :>

`Param` should use `ApplicationCommandInteraction[BotT]` with e.g. a `BotT = TypeVar("BotT", bound=AnyBot)`, that way it won't resolve to an unknown type in lambdas like this one.